drivers/libvirt/libvirt_linux_amd64.go

```diff
@@ -0,0 +1,695 @@
+		cli.StringFlag{
+			EnvVar: "LIBVIRT_BOOT2DOCKER_URL",
+			Name:   "libvirt-boot2docker-url",
+			Usage:  "The URL of the boot2docker image. Defaults to the latest available version",
+			Value:  "",
+		},
+	}
+}
+
+func NewDriver(machineName string, storePath string, caCert string, privateKey string) (drivers.Driver, error) {
+	d := Driver{MachineName: machineName, storePath: storePath, CaCertPath: caCert, PrivateKeyPath: privateKey}
+	conn, err := libvirt.NewVirConnection(d.connectionString)
+	if err != nil {
+		log.Warnf("Failed to connect: %s", err)
+		return nil, err
+	}
+	d.conn = &conn
+	return &d, nil
+}
+
+func (d *Driver) DriverName() string {
+	return "libvirt"
+}
+
+func (d *Driver) GetURL() (string, error) {
+	log.Debugf("GetURL called")
+	ip, err := d.GetIP()
+	if err != nil {
+		log.Warnf("Failed to get IP: %s", err)
+		return "", err
+	}
+	if ip == "" {
+		return "", nil
+	}
+	return fmt.Sprintf("tcp://%s:2376", ip), nil
+}
+
+func (d *Driver) SetConfigFromFlags(flags drivers.DriverOptions) error {
+	log.Debugf("SetConfigFromFlags aclled")
+	d.Hypervisor = flags.String("libvirt-hypervisor")
+	d.Memory = flags.Int("libvirt-memory")
+	d.DiskSize = flags.Int("libvirt-disk-size")
+	d.VCPU = flags.Int("libvirt-vcpu")
+	d.Network = flags.String("libvirt-network")
+	d.Boot2DockerURL = flags.String("libvirt-boot2docker-url")
+	d.connectionString = flags.String("libvirt-connection")
+	return nil
+}
+
+func (d *Driver) validateNetwork(name string) error {
+	log.Debugf("Validating network %s", name)
+	network, err := d.conn.LookupNetworkByName(name)
+	if err != nil {
+		log.Warnf("Unable to locate network %s", name)
+		return err
+	}
+	xmldoc, err := network.GetXMLDesc(0)
+	if err != nil {
+		return err
+	}
+	// XML structure:
+	//  <network>
+	//      ...
+	//      <forward mode='nat'>
+	//      <ip address='a.b.c.d' netmask='255.255.255.0'>
+	//          <dhcp>
+	//              <range start='a.b.c.d' end='w.x.y.z'/>
+	//          </dhcp>
+	type Forward struct {
+		Mode string `xml:"mode,attr"`
+	}
+	type Ip struct {
+		Address string `xml:"address,attr"`
+		Netmask string `xml:"netmask,attr"`
+	}
+	type Network struct {
+		Forward Forward `xml:"forward"`
+		Ip      Ip      `xml:"ip"`
+	}
+
+	var nw Network
+	err = xml.Unmarshal([]byte(xmldoc), &nw)
+	if err != nil {
+		return err
+	}
+	switch nw.Forward.Mode {
+	case "nat":
+		fallthrough
+	case "route":
+		fallthrough
+	case "": // No forwarding
+		if nw.Ip.Address == "" {
+			log.Warnf("Network doesn't have valie IP range")
+			return errors.New("Unsupported libvirt network mode")
+		}
+	case "bridge":
+		log.Infof("Network %s bridged", name)
+		/*
+		 * TODO
+		 *  There isn't really a silver bullet for bridged networks.
+		 *  Since the DHCP server is most likely not on this box,
+		 *  we can't look up the IP address, and our ARP
+		 *  table probably wont have the MAC listed.  To support
+		 *  this we can ask the user to give us the IP.
+		 */
```

Comment:
does this mean you're only creating one network  for the vm? B2D uses 2 to get around this problem - the first is a NATed one so that the cli can ask the vm for the current IP of the more general one (and at the same time, ask about the Docker daemon port and other configurables.